ui/tests/integration/components/sync-status-badge-test.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import hbs from 'htmlbars-inline-precompile';
+import { render } from '@ember/test-helpers';
+import { PAGE } from 'vault/tests/helpers/sync/sync-selectors';
+import { toLabel } from 'core/helpers/to-label';
+
+module('Integration | Component | SyncStatusBadge', function (hooks) {
+  setupRenderingTest(hooks);
+
+  const SYNC_STATUSES = {
+    SYNCING: { icon: 'sync', color: 'neutral' },
+    SYNCED: { icon: 'check-circle', color: 'success' },
+    UNSYNCING: { icon: 'sync-reverse', color: 'neutral' },
+    UNSYNCED: { icon: 'sync-alert', color: 'warning' },
+    INTERNAL_VAULT_ERROR: { icon: 'x-circle', color: 'critical' },
+    CLIENT_SIDE_ERROR: { icon: 'x-circle', color: 'critical' },
+    EXTERNAL_SERVICE_ERROR: { icon: 'x-circle', color: 'critical' },
+    UNKNOWN: { icon: 'help', color: 'neutral' },
+  };
```

Comment:
Thinking back to our discussion and I see now the benefits of having this status data in a util exported somewhere. I still think it's fine for now but something to keep in mind if we are needing to duplicate this.